app/models/user.rb

```diff
@@ -0,0 +1,37 @@
+class User < ApplicationRecord
+  has_secure_password
+
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+
+  validates :name, presence: true,
+    length: {maximum: Settings.user.max_name_length}
+
+  validates :email, presence: true,
+    length: {maximum: Settings.user.max_email_length},
+    format: {with: VALID_EMAIL_REGEX}, uniqueness: {case_sensitive: false}
+
```

Comment:
```suggestion
  validates :name, presence: true,
    length: {maximum: Settings.user.max_name_length}
  validates :email, presence: true,
    length: {maximum: Settings.user.max_email_length},
    format: {with: VALID_EMAIL_REGEX}, uniqueness: {case_sensitive: false}
```
2 validates cùng loại này thì có thể k cần line trống, chủ yếu để phân cách các thành phần khác loại